B-2

AlexeyVolkov2023 opened the pull request:
Создание классов волонтера, животного, породы, вида, также сделал несколько VO

AlexeyVolkov2023 committed:
B-2

Создание классов волонтера, животного, породы, вида, также сделал несколько VO

KirillSachkov commented on backend/src/PetHelp.Domain/AnimalManagement/AggregateRoot/Volunteer.cs:
```diff
@@ -0,0 +1,45 @@
+        string phoneNumber,
+        List<SocialNetwork> socialNetworks,
+        List<PaymentDetails> paymentDetails,
+        List<Pet> ownedPets) : base(id)
```
при создании волонтера мы не должны сразу заполнять животных

KirillSachkov commented on backend/src/PetHelp.Domain/AnimalManagement/VO/PetStatus.cs:
```diff
@@ -0,0 +1,19 @@
+﻿namespace PetHelp.Domain.AnimalManagement.VO;
+
+public record PetStatus
```
все VO лучше наследовать от класса ComparableValueObject из библиотеки CsharpfunctionalExtensions

AlexeyVolkov2023 committed:
B-2 refactoring

Добавил наследование VO от ComparableValueObject, в волонтере убрал добавление коллекции Pet в конструкторе, сделал для этого отдельный метод
